### PR TITLE
Add Writer::to_string and Writer::to_bytes methods

### PR DESCRIPTION
**Pull request by Xion**
Docs on `Writer::from_memory` mention the existence of `to_string` and `to_bytes` methods that can be used to retrieve the encoded data at any time. Unfortunately, those methods don't actually exist.

This PR adds them. Both consume the `Writer<Vec<u8>>` and produce either a `String` or the underlying `Vec<u8>` without copying.

### CODE REVIEW

File: src/writer.rs [hunk 1]
```diff
@@ -143,6 +143,24 @@ impl Writer<Vec<u8>> {
             Ok(()) => &**self.buf.get_ref(),
         }
     }
+
+    /// Returns the written CSV data as owned string.
+    pub fn to_string(self) -> String {
```
Comment (BurntSushi) [inline src/writer.rs]: Could you rename this to `into_string`? (Per conventions in `std`.) Similarly for `to_bytes` -> `into_bytes`.

**Comment by BurntSushi** [comment]
The documentation for `from_memory` should say `as_bytes`/`as_string`, not `to_bytes`/`to_string`.

**Comment by Xion** [comment]
Fixed both.

**Comment by BurntSushi** [comment]
Great, thank you!

**Comment by BurntSushi** [comment]
This PR is now on crates.io in `0.14.4`.

**Pull request merged**